src/rust/pants_ng/options/src/pants_invocation.rs

```diff
@@ -0,0 +1,215 @@
+// Copyright 2025 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+use std::env;
+use std::iter::Peekable;
+use std::path;
+use std::sync::LazyLock;
+
+use regex::Regex;
+
+static NAME_RE: LazyLock<Regex> = LazyLock::new(|| Regex::new(r"^[a-z|A-Z]\w*$").unwrap());
```

Comment:
Normally I use `[a-zA-Z]` in python regex. Does rust use `|` in character groups to mean "or"? Or is this saying that `|` is a valid first character of a name?